cloudant-sync-datastore-core/src/main/java/com/cloudant/common/SimpleChangeNotifyingMap.java

```diff
@@ -1,48 +1,54 @@
+/*
+ * Copyright © 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
 package com.cloudant.common;
 
-import com.google.common.collect.ForwardingMap;
-
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * Created by tomblench on 10/02/16.
+ *
  * @api_private
  */
-public class SimpleChangeNotifyingMap<K, V> extends ForwardingMap<K, V> {
-
-    Map<K, V> delegateMap;
-    private boolean mapChanged;
-
-    public SimpleChangeNotifyingMap(Map<K, V> delegateMap) {
-        this.delegateMap = delegateMap;
-        this.mapChanged = false;
-    }
-
-    protected Map<K, V> delegate() {
-        return delegateMap;
-    }
-
-    public boolean hasChanged() {
-        return mapChanged;
+public class SimpleChangeNotifyingMap {
```

Comment:
OK I think this approach is worth sticking with. Is there a reason for `isChanged` to be an `AtomicBoolean`?